Day_3/day_3.py

```diff
@@ -0,0 +1,428 @@
+        """
+        Calculate the long range tail correction for a system.
+
+        Parameters
+        ----------
+        box_length : float   
+            The length of the simulation box
+        cutoff : float
+            The simulation cutoff. Beyond this distance, interactions are not calculated.
+        number_particles:
+            The number of particles in the simulation
+        
+        Returns
+        -------
+        e_correction : float
+            The long distance tail correction.
+        """
+
+        volume = self.box.volume
+        sig_by_cutoff3 = np.power(1.0 / self.cutoff, 3)
+        sig_by_cutoff9 = np.power(sig_by_cutoff3, 3)
+        e_correction = sig_by_cutoff9 - 3.0 * sig_by_cutoff3
+        e_correction *= 8.0 / 9.0 * np.pi * self.box.number_particles / volume * self.box.number_particles
+
+        self.tail_correction = e_correction
+    
+    def calculate_total_energy(self):
+        self.calculate_tail_correction()
+        self.calculate_total_pair_energy()
+        self.total_energy = self.total_pair_energy + self.tail_correction
+    
+    def get_particle_energy(self, i_particle, particle_movement=None):
+        """
+        Calculate the interaction energy of a particle with its environment (all other particles in the system)
+
+        Parameters
+        ----------
+        i_particle : int
+            The particle index for which to calculate the energy
+       
+        Returns
+        -------
+        e_total : float 
+            The pairwise interaction energy of he i_th particle with all other particles in the system.
+        """
+
+        calculate_coordinates = self.coordinates.copy()
+
+        if particle_movement is not None:
+            calculate_coordinates[i_particle] += particle_movement
+
+        e_total = 0.0
+        i_position = calculate_coordinates[i_particle]
+
+        cutoff2 = self.cutoff ** 2
+        particle_count = self.box.number_particles
+
+        for j_particle in range(particle_count):
+            if i_particle != j_particle:
+
+                j_position = calculate_coordinates[j_particle]
+                rij2 = self.box.minimum_image_distance(i_position, j_position)
+
+                if rij2 < cutoff2:
+                    e_pair = lennard_jones_potential(rij2)
+                    e_total += e_pair
+
+        return e_total
+
+    
+def generate_initial_coordinates(method, **kwargs):
+    """
+    Generate initial coordinates for MC system.
+
+    Parameters
+    ----------
+    method : str
+        Method for creating initial configuration. Valid options are 'random' or 'file'. 'random' will place particles in the simulation box randomly, depending on the num_particles argument, while 'file' will read coordinates from an xyz file.
+
+    Returns
+    -------
+    coordinates : numpy array
+        Array of coordinates (x, y, z)
+    """
+    
+    coord_method = get_method(method)
+    return coord_method(**kwargs)
+
+def get_method(method):
```

Comment:
This should move to the `generate_initial_coordinates` function. The rule for creating functions is we shouldn't create a function if it's super short and is used in only one place in the code.
We would do this if the original method is long and complex and we are breaking down the code into pieces, otherwise it's a wrong practice.